### PR TITLE
feat(engine/image): implement the shadow effect.

### PR DESCRIPTION
**Pull request by fky2015**
output

### CODE REVIEW

File: engine/src/image.rs [hunk 1]
```diff
@@ -8,18 +8,48 @@ use super::svg::SvgTangibleObject;
...
+        Self {
+            path,
+            round: None,
+            shadow: None,
+        }
+    }
+
+    /// Get padding size.
+    ///
+    /// Padding size is the extra space around the image,
```
Comment (phy25) [inline engine/src/image.rs:26]: nitpick: might be worth calling out that the return value is on x and y and we treat both side on one direction the same spacing

File: engine/src/image.rs [hunk 2]
```diff
@@ -40,24 +70,59 @@ impl SvgTangibleObject for Image {
...
+            }
+
+            if let Some((offset, std_deviation)) = self.shadow {
+                let mut filter = Element::new("filter");
+                filter.set_attr("id", "shadow");
+                let mut drop_shadow = Element::new("feDropShadow");
+                drop_shadow.set_attr("dx", offset.0.to_string());
+                drop_shadow.set_attr("dy", offset.1.to_string());
+                drop_shadow.set_attr("stdDeviation", std_deviation.to_string());
+                drop_shadow.set_attr("flood-opacity", "0.6");
```
Comment (phy25) [inline engine/src/image.rs:98]: I created a new task to support customizing this and `flood-color`, it’s now black by default https://linear.app/footlights/issue/FL-28

File: engine/src/image.rs [hunk 3]
```diff
@@ -8,18 +8,48 @@ use super::svg::SvgTangibleObject;
...
+                offset.1 + 3 * std_deviation + 1,
+            )
+        } else {
+            (0, 0)
+        }
     }
 }
 
 impl SizeOptionT for Image {
     fn get_size_option(&self) -> SizeOption {
```
Comment (phy25) [inline engine/src/image.rs:46]: Need to have test coverage on expanded `get_size_option` in the future as it’s highly coupled with the `to_svg` behavior.

File: engine/src/image.rs (hunk 2 above)
Comment (fky2015) [reply engine/src/image.rs:156]: We'll commence this issue when the structure is more stable.

**Pull request merged**